### DISCUSSION TITLE
OWL software not working for me

### DISCUSSION THREAD

**Discussion by PatHulme** [geezacoleman/OpenWeedLocator#74] [Q&A] [answered]
Not sure if this is the correct forum.  However.
First - I'm a beginner at building electronic devices - but have done a bit of programming
I've built 2 OWL boxes over the past 8 months following the excellent description on Github
Now I'm trying to load the software using the Quick Method as described on Github as I don't have skills to connect the boards to the internet 
I have followed the same process for the 2 boxes on 2 sd cards (one stores 32 and the other 64GB) so 4 times.  This is:
- Flash software to card and place in holder in raspberry pi
- Power up board and screen displays Open Weed Locator 
- In the terminal window - owl does not appear before pi@raspberrypi:~
- owl does appear at the start of each line if I change to the owl virtual environment with the command workon owl
- When I enter ~owl/./owl.py -- show-display, the computer message is:
   bash: /home/owl/.owl/owl.py No such file or directory

I agree with the computer in that owl .py is not in the owl folder, neither is greenongreen.py
![image](https://user-images.githubusercontent.com/48673696/232382516-2a93f2a1-5d24-4774-8677-0328e6716a2e.png)
To be confident I'm using current software I've downloaded the latest software using the link in the instructions.
The owl.img file is dated in the zip file is dated 8/07/2021 5:52PM
Could you please help as I have done all I can think of

**Comment by geezacoleman** [comment, marked as answer]
Thanks for the question - looks like you've found some errors I've overlooked. 

First, is that for this release of the OWL software with the quick install method, it uses the old `greenonbrown.py` not the combined `owl.py` as it is now. So to run it you need to type `./greenonbrown.py`.

To fix it , change this line:
```
(owl) pi@raspberrypi:~ $ ~/owl/./owl.py --show-display
```

to:
```
(owl) pi@raspberrypi:~ $ cd owl 
(owl) pi@raspberrypi:~ /owl $ ./greenonbrown.py  
```
There is a bug in this original version of the code that will cause an error if you run it from outside of the `owl` directory.

It is running the oldest version of the OWL software (we really need to update this sorry!), so you'll need to open up `greenonbrown.py` as a file and change the settings at the bottom of the code. If you want it to run automatically you need to change `headless=True` to `headless=False`. I would also recommend updating the software if possible - the performance gain and bug fixes is considerable (~20% in recall) from the changes we've made over the last couple of years. Uploading a new disk image is also a high priority on the to do list, which would fix all these issues.

I will update the instructions based on your information so people know what to do if they have the older software.

Let me know how this goes for you! Post any further issues up here too and I can try addressing them.

**Comment by PatHulme** [comment]
Guy

Thanks heaps for the guidance

Another step on the journey completed.

 

Last night I had worked out the that I should follow last year’s instructions, but I encountered the bug that you mentioned. 

 

I have scoured the github site and cant find a more recent version of the img file than the screen capture below.  I thought that I might be able to copy that python scripts onto the sd card.  But windows 10 cant read the partition.   So I’ll make do with your original software for now.

 

I think I need to work out how to get the raspberry pi to talk to the internet so I can use the detailed installation method and download the updated owl files onto the raspberry pi

 

Pat

**Reply by geezacoleman** [reply]
Unfortunately that's the only version of the software that exists currently as an image ready for download. It's something that's on the list of things to do. Will let you know when we get it up!

In the mean time, I reckon the best way to get the new python software onto the Pi without setting up an internet connection will be to use a USB drive, download this repository from Github on your main computer and then copy it to the Pi.

Otherwise, I've found the easiest way to set up a connection for the Pi is just to use your phone as a wifi hot spot if you have enough coverage.

**Comment by PatHulme** [comment]
Guy,

I’m sort of making progress, but I’m way outside my comfort zone.  I’m sort of busy and work on this owl project every now and then.

 

I have built 2 owl units (and bought the solenoids).

 

Last night I had them both working in that they would image a green disk I have and get an image, and  close the appropriate relay.

 

I was excited.

 

Then they stopped working.   Tonight I had one working and the camera was working on the second (I saw image on screen), but not sending message to relay.   I can work on this.

 

BUT software 

 

I downloaded the updated firmware and have loaded it twice.

 

Once using a USB as you suggested, and the second time via wi-fi which is brilliant.

 

But I don’t think the new script is running  Random question -  is there a pi command similar to ctrl alt delete in windows to open a task manager

 

Now the software challenge

I can only get the OWL software to work with a workaround

 

I boot the raspberry pi and the OWL screen comes up, but nothing else happens

I tried the ps – C greenonbrown.py command and don’t get a number under tty

 

Then I go through

workon owl

cd ~/owl

~/owl/./greenonbrown.py –show-display

 

And the raspberry pi doesn’t whinge, but doesn’t do anything either

                                                                                                                                                                                                

I can cheat to get the  software working in that I can get the system to start by using the mouse, clicking on the old grrenonbrown.py script and select execute it in window

 

The ps -C greenonbrown.py command then returns a PID number and 

The sudo kill XXX command stops the program

 

In summary, the hardware seems to work – so you have damn fine instructions for that

But I cant get the software to work on startup

 

I changed the headless command in the old script and it works if I click on the script and tell the raspberry pi to execute it.

 

I thought that I would be cunning and rename the folder containing the original OWL script to owl and reboot the owl Again the owl screen, but nothing else.

 

So, sadly I need help again

 

I will try to flash another card and start again, but there are really basic things I’m not sure of  - like should I reformat the mini SD card before I flash the software onto it.

Sorry

 

Pat

**Reply by geezacoleman** [reply]
Hi Pat,

Thanks for persisting with these issues! Biggest problem I see is from us changing from running `greenonbrown.py` to `owl.py` and not documenting that clearly enough. It was done to incorporate the 'Green on Green' feature and make the code more modular/useable for adding new features. But ended up causing more in different places! I'll try and make that more clear. Also have a bit of time to work on OWL things this weekend, so I'll make it a priority to update the system image we have there so it can simply be downloaded and run straight away with the latest software.

To fix your problem, another OWL user has had similar issues caused by the greenonbrown/owl.py difference, so check out [the response to that](https://github.com/geezacoleman/OpenWeedLocator/issues/78#issuecomment-1565434369). It should hopefully solve your issues. It makes the most of wi-fi access to update the OWL. I have also made new update_owl.sh scripts that automates much of the process.

While it works, the software on the image you can download is quite buggy and slow, so I wouldn't recommend using it longer term without updating, using the method linked above.

**Comment by PatHulme** [comment]
Guy

Don’t worry

 

I flashed another minSD and it worked straight away

I will go through the process of upgrading  the software next week

 

I will work on some other things for now – like a trailer with clearance in front, plumbing, loom etc,

 

Pat

 

From: Pat ***@***.***> 
Sent: Thursday, 6 July 2023 7:26 PM
To: 'geezacoleman/OpenWeedLocator' ***@***.***>
Subject: RE: [geezacoleman/OpenWeedLocator] OWL software not working for me (Discussion #74)

 

Guy,

I’m sort of making progress, but I’m way outside my comfort zone.  I’m sort of busy and work on this owl project every now and then.

 

I have built 2 owl units (and bought the solenoids).

 

Last night I had them both working in that they would image a green disk I have and get an image, and  close the appropriate relay.

 

I was excited.

 

Then they stopped working.   Tonight I had one working and the camera was working on the second (I saw image on screen), but not sending message to relay.   I can work on this.

 

BUT software 

 

I downloaded the updated firmware and have loaded it twice.

 

Once using a USB as you suggested, and the second time via wi-fi which is brilliant.

 

But I don’t think the new script is running  Random question -  is there a pi command similar to ctrl alt delete in windows to open a task manager

 

Now the software challenge

I can only get the OWL software to work with a workaround

 

I boot the raspberry pi and the OWL screen comes up, but nothing else happens

I tried the ps – C greenonbrown.py command and don’t get a number under tty

 

Then I go through

workon owl

cd ~/owl

~/owl/./greenonbrown.py –show-display

 

And the raspberry pi doesn’t whinge, but doesn’t do anything either

                                                                                                                                                                                                

I can cheat to get the  software working in that I can get the system to start by using the mouse, clicking on the old grrenonbrown.py script and select execute it in window

 

The ps -C greenonbrown.py command then returns a PID number and 

The sudo kill XXX command stops the program

 

In summary, the hardware seems to work – so you have damn fine instructions for that

But I cant get the software to work on startup

 

I changed the headless command in the old script and it works if I click on the script and tell the raspberry pi to execute it.

 

I thought that I would be cunning and rename the folder containing the original OWL script to owl and reboot the owl Again the owl screen, but nothing else.

 

So, sadly I need help again

 

I will try to flash another card and start again, but there are really basic things I’m not sure of  - like should I reformat the mini SD card before I flash the software onto it.

Sorry

 

Pat

**Comment by PatHulme** [comment]
Guy,

Thank you very much for your prompt reply.

I’ve been on the road for the past 3 days, and will try to implement your recommendations tonight.

I’ve also been spreading the word about me trying to build a camera spray.

Just say that people I’ve described the project to are watching with interest.  So I guess that means I have some responsibility to you to do my best to make it work

Pat

**Comment by david-j-webster** [comment]
I have been trying to load the software on both a Rpi3 and a Jetson Nano and i must be missing something I can not get them to work. I have tried both the quick way and the step by step on the raspberry pi

**Reply by geezacoleman** [reply]
Assuming this is related to your other comment on #17, I imagine it's the picamera version issue. My response there should hopefully clarify things.
